packages/TorneloScoresheet/src/pages/EnterPgn/EditCurrentEvent.tsx

```diff
@@ -0,0 +1,182 @@
+  display: boolean;
+  dismiss: () => void;
+};
+const EditCurrentEvent: React.FC<EditCurrentEventParams> = ({
+  dismiss,
+  display,
+}) => {
+  const [arbiterInfo, setArbiterInfo] = useArbiterInfo();
+  const [, showError] = useError();
+
+  const [inScanQRMode, setInScanQRMode] = useState(true);
+
+  const [arbiterPin, setArbiterPin] = useState(arbiterInfo?.pin ?? '');
+  const [pgnUrl, setPgnUrl] = useState(arbiterInfo?.broadcastUrl ?? '');
+  const [arbiterEmailSecret, setArbiterEmailSecret] = useState(
+    arbiterInfo?.arbiterToken ?? '',
+  );
+  const [divisionId, setDivisionId] = useState(arbiterInfo?.divisionId ?? '');
+  const [arbiterId, setArbiterId] = useState(arbiterInfo?.userId ?? '');
+  const [eventName, setEventName] = useState(arbiterInfo?.eventName ?? '');
+  const [arbiterName, setArbiterName] = useState(arbiterInfo?.userName ?? '');
+
+  const inputBoxesContent: [
+    string,
+    React.Dispatch<React.SetStateAction<string>>,
+    string,
+  ][] = [
+    [pgnUrl, setPgnUrl, 'Tournament Url'],
+    [arbiterPin, setArbiterPin, 'Arbiter Pin'],
+    [arbiterEmailSecret, setArbiterEmailSecret, 'Arbiter Email Secret'],
+    [divisionId, setDivisionId, 'Division Id'],
+    [arbiterId, setArbiterId, 'Arbiter Id'],
+    [arbiterName, setArbiterName, 'Arbiter Name'],
+    [eventName, setEventName, 'Event Name'],
+  ];
+
+  const validateArbiterInfo = (arbiterInfo: ArbiterInfo): boolean => {
+    if (arbiterInfo.arbiterToken == '') {
+      showError('Arbiter Email token cannot be blank');
+      return false;
+    }
+    if (arbiterInfo.divisionId == '') {
+      showError('Division Id cannot be blank');
+      return false;
+    }
+    if (arbiterInfo.eventName == '') {
+      showError('Event Name cannot be blank');
+      return false;
+    }
+    if (arbiterInfo.userName == '') {
+      showError('Arbiter Name cannot be blank');
+      return false;
+    }
+    if (arbiterInfo.userId == '') {
+      showError('Arbiter Id cannot be blank');
+      return false;
+    }
+    if (arbiterInfo.pin == '') {
+      showError('Arbiter Pin cannot be blank');
+      return false;
+    }
+    if (!validUrl(arbiterInfo.broadcastUrl)) {
+      showError('Invalid Pgn Url');
+      return false;
+    }
+    return true;
+  };
+
+  const confirmArbiterInfo = async (): Promise<void> => {
+    const newArbiterInfo: ArbiterInfo = {
+      pin: arbiterPin,
+      broadcastUrl: pgnUrl,
+      arbiterToken: arbiterEmailSecret,
+      divisionId,
+      userId: arbiterId,
+      userName: arbiterName,
+      eventName,
+    };
+    dismiss();
+
+    if (validateArbiterInfo(newArbiterInfo)) {
+      await setArbiterInfo(newArbiterInfo);
+    }
+  };
+  return (
+    <>
+      <Sheet title="Edit Event" dismiss={dismiss} visible={display}>
+        <View style={styles.editEventHeader}>
+          <TextIconButton
+            Icon={inScanQRMode ? ICON_TEXT : ICON_QR}
+            text={inScanQRMode ? 'Text input' : 'QR input'}
```

Comment:
Could these please be Text Input and QR Input? I think it looks a bit nicer than the lower case i